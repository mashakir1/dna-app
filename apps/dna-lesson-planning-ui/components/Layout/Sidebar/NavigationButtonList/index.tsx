// Imports
import { useRouter } from "next/router";
import useTranslation from "next-translate/useTranslation";

// Components
import { IconLinkButton } from "components/Buttons";

// Icons
import { FileIcon, BarChartIcon } from "components/Svg";

// Styles
import styles from "styles/components/sidebar/navigationButton.module.scss";

const buttonList = [
  {
    text: "my-lessons",
    icon: <FileIcon />,
    href: "/"
  },
  {
    text: "lesson-data",
    icon: <BarChartIcon />,
    href: "/data"
  }
];

const NavigationButtonList: () => JSX.Element = () => {
  const { pathname } = useRouter();
  const { t } = useTranslation("common");
  return (
    <>
      {buttonList.map(({ text, icon, href }) => (
        <IconLinkButton
          key={text}
          text={t(text)}
          icon={icon}
          href={href}
          classNames={`${styles.sidebarButton} ${
            pathname === href ? styles.highlighted : ""
          } ${styles.navigationButton} flex items-center `}
        />
      ))}
    </>
  );
};

export default NavigationButtonList;
