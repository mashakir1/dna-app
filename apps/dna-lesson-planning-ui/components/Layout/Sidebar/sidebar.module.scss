.sidebar {
  padding: $spacing-l $spacing-m;
  background-color: $colour_navy;
  height: 100vh;
  width: 20%;
}

.sidebarContent {
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: column;
}
