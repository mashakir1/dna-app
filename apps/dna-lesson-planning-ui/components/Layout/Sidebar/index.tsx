// * Imports
import useTranslation from "next-translate/useTranslation";

//* Components
import { NordAngliaLogo, SpeechBubbleIcon } from "components/Svg";
import Spacing from "components/Layout/Spacing";
import NavigationButtonList from "./NavigationButtonList";
import { IconLinkButton } from "components/Buttons";

//* Styles
import sidebarStyling from "./sidebar.module.scss";
import feedbackStyling from "styles/components/sidebar/feedback.module.scss";
import navigationButtonStyling from "styles/components/sidebar/navigationButton.module.scss";

const Sidebar: () => JSX.Element = () => {
  const { t } = useTranslation("common");
  return (
    <div className={sidebarStyling.sidebar}>
      <div className={sidebarStyling.sidebarContent}>
        <NordAngliaLogo />
        <Spacing height={"xxl"} />
        <NavigationButtonList />
        <IconLinkButton
          href="/feedback"
          icon={<SpeechBubbleIcon />}
          text={t("feedback")}
          classNames={`${navigationButtonStyling.sidebarButton} ${feedbackStyling.feedback} flex justify-center items-center`}
        />
      </div>
    </div>
  );
};

export default Sidebar;
