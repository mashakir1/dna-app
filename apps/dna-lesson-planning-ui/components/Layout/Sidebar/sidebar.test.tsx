import { render } from "@testing-library/react";
import Sidebar from ".";

jest.mock("components/Svg", () => ({
  NordAngliaLogo() {
    return <div>NordAngliaLogo</div>;
  },
  SpeechBubbleIcon() {
    return <div>SpeechBubbleIcon</div>;
  }
}));

const navigationButtonListMock = () => (<div>NavigationButtonList</div>);
jest.mock("components/Layout/Sidebar/NavigationButtonList", () => navigationButtonListMock);

const spacingMock = () => {
  return <div>Spacing</div>;
}
jest.mock("components/Layout/Spacing", () => spacingMock);

jest.mock("components/Buttons", () => ({
  IconLinkButton() {
    return <div>IconLinkButton</div>;
  }
}));

describe("Sidebar", () => {
  it("renders correctly", () => {
    expect(render(<Sidebar />).container).toMatchSnapshot();
  });
});



