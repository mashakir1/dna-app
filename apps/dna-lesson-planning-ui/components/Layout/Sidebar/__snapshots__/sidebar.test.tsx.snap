// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders correctly 1`] = `
<div>
  <div
    class="sidebar"
  >
    <div
      class="sidebarContent"
    >
      <div>
        NordAngliaLogo
      </div>
      <div>
        Spacing
      </div>
      <div>
        NavigationButtonList
      </div>
      <div>
        IconLinkButton
      </div>
    </div>
  </div>
</div>
`;
